lot.php

```diff
@@ -2,21 +2,58 @@
 
 require_once('boot.php');
 
+$sqlCategories = getCategories($link);
 $userName = getSessionName();
-$lotId = intval($_GET['ID']);
+$errors = '';
 
-$sqlCategories = getCategories($link);
+$lotId = (int)$_GET['ID'];
+
+$sqlCatLot = getCatLotMaxPrice($link, $lotId);
+if (checkExistDbVal($sqlCatLot)) {
+    $pageContent = include_template('404.php', [
+    'sqlCategories' => $sqlCategories,
+    'userName' => $userName,
+    ]);
+    $layoutContent = include_template('layout.php', [
+        'categories' => $sqlCategories,
+        'content' => $pageContent,
+        'title' => 'Страница не найдена',
+        'userName' => $userName,
+    ]);
 
-$sqlCatLot = getCatLot($link, $lotId);
-checkExistDbVal($sqlCatLot);
+    print($layoutContent);
+    exit();
+}
 
 $sqlBidUser = getBidUser($link, $lotId);
 
+$price = getPrice($sqlBidUser, $sqlCatLot);
+
+if ($_SERVER['REQUEST_METHOD'] === 'POST') { 
+    if (empty($userName)) {     //Проверка, что пользователь залогинен.
+        header("Location: login.php");
+        exit();
+    }
+    $userPrice = trim(filter_input(INPUT_POST, 'cost')); // Получаем значение из формы и проверяем что значение целое
+
+    $errors = validateFormLot($userPrice, $price);
+
+    if(empty($errors)) {
+       if (!$res = addBid($link, $lotId, $userPrice)) {
+           print("Error MySQL: " . $error);
+           exit;
+       }
+       header("Location: /lot.php?ID=".$lotId."");
```

Comment:
Тут можно так, например 
```suggestion
       header("Location: /lot.php?ID={$lotId}");
```

Используй двойные кавычки там, где предполагается переменная внутри строки